### PR TITLE
CVSL-1740: Pull in changes from template project

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,49 +1,53 @@
...
 WORKDIR /app
 
-# Cache breaking
-ENV BUILD_NUMBER ${BUILD_NUMBER:-1_0_0}
-ENV GIT_REF ${GIT_REF:-xxxxxxxxxxxxxxxxxxx}
-    
+# Cache breaking and ensure required build / git args defined
+RUN test -n "$BUILD_NUMBER" || (echo "BUILD_NUMBER not set" && false)
+RUN test -n "$GIT_REF" || (echo "GIT_REF not set" && false)
+RUN test -n "$GIT_BRANCH" || (echo "GIT_BRANCH not set" && false)
```
Comment (andrewrlee) [inline Dockerfile:21]: changes from template - the info endpoint will be a closer to spring boot and include info about git branch (which will generally be main for us!)

File: helm_deploy/create-and-vary-a-licence/values.yaml [hunk 2]
```diff
@@ -28,8 +29,6 @@ generic-service:
     hosts:
       - app-hostname.local # override per environment
     tlsSecretName: create-and-vary-a-licence-cert
-    v1_2_enabled: true
-    v0_47_enabled: false
```
Comment (andrewrlee) [inline helm_deploy/create-and-vary-a-licence/values.yaml:32]: This is default behaviour now

File: helm_deploy/create-and-vary-a-licence/values.yaml [hunk 3]
```diff
@@ -17,6 +17,7 @@ generic-service:
   nameOverride: create-and-vary-a-licence
   replicaCount: 4
   serviceAccountName: create-and-vary-a-licence
+  productId: "DPS011"
```
Comment (andrewrlee) [inline helm_deploy/create-and-vary-a-licence/values.yaml:20]: This will be exposed by the info endpoint

File: feature.env [hunk 4]
```diff
@@ -16,7 +16,7 @@ SYSTEM_CLIENT_SECRET=clientsecret
 SQS_POLLING_WAIT_TIME_MS=2000
 REDIS_HOST=localhost
 HARD_STOP_ENABLED=true
-
+TOKEN_VERIFICATION_ENABLED=true
```
Comment (andrewrlee) [inline feature.env:19]: tests now use token verification, the health checks now check partial failure works as expected

File: integration_tests/mockApis/auth.ts [hunk 5]
```diff
@@ -103,29 +104,9 @@ const token = (authorities: string[], authSource: string) =>
       },
     },
   })
-
-const stubUserEmail = () =>
```
Comment (andrewrlee) [inline integration_tests/mockApis/auth.ts:107]: moved this into dedicated manage-users api file

File: jobs/deactivateReleaseDatePassedLicences.ts [hunk 6]
```diff
@@ -3,10 +3,10 @@
  * Do appinsights first to instrument the logger
  */
 import 'reflect-metadata'
-import { initialiseAppInsights, buildAppInsightsClient, flush } from '../server/utils/azureAppInsights'
+import { initialiseAppInsights, flush } from '../server/utils/azureAppInsights'
+import applicationInfo from '../server/applicationInfo'
 
-initialiseAppInsights()
-buildAppInsightsClient('create-and-vary-a-licence-deactivate-release-date-passed-licences-job')
+initialiseAppInsights(applicationInfo('create-and-vary-a-licence-deactivate-release-date-passed-licences-job'))
```
Comment (andrewrlee) [inline jobs/deactivateReleaseDatePassedLicences.ts:9]: simplified this set up a bit - the buildAppInsightsClient was really "configure default client that is used for everything" and we didn't actually need a reference to it, so merged the two.

File: server.ts [hunk 7]
```diff
@@ -3,16 +3,20 @@
  * Do appinsights first as it does some magic instrumentation work, i.e. it affects other 'require's
  * In particular, applicationinsights automatically collects bunyan logs
  */
-import { initialiseAppInsights, buildAppInsightsClient } from './server/utils/azureAppInsights'
+import { initialiseAppInsights } from './server/utils/azureAppInsights'
+import createApplicationInfo from './server/applicationInfo'
 
-initialiseAppInsights()
-buildAppInsightsClient()
+const applicationInfo = createApplicationInfo()
```
Comment (andrewrlee) [inline server.ts:9]: creating this here and then using dependency injection to pass through to where needed

File: server/applicationInfo.ts [hunk 8]
```diff
@@ -2,11 +2,30 @@ import fs from 'fs'
 import path from 'path'
 import config from './config'
 
-const packageJson =
```
Comment (andrewrlee) [inline server/applicationInfo.ts:5]: there was some hackiness here as during the relative location of package.json would vary dependent on whether we were running tests (compiled on fly) vs running the app (compiled to dist folder)

This is no longer needed due to use of DI

File: server/applicationInfo.ts [hunk 9]
```diff
@@ -2,11 +2,30 @@ import fs from 'fs'
...
+const getNameFromPackageJson = () => {
+  const packageJson = path.join(__dirname, '../../package.json')
+  const { name } = JSON.parse(fs.readFileSync(packageJson).toString())
+  return name
+}
 
-export default { applicationName, buildNumber, gitRef, gitShortHash: gitRef.substring(0, 7) }
+export default (overrideName?: string): ApplicationInfo => {
+  return {
+    applicationName: overrideName || getNameFromPackageJson(),
```
Comment (andrewrlee) [inline server/applicationInfo.ts:24]: little optimisation to only load name from the file system if no override was specified

File: server/data/healthCheck.test.ts [hunk 10]
```diff
@@ -29,7 +29,7 @@ describe('Service healthcheck', () => {
       fakeServiceApi.get('/ping').reply(200, 'pong')
 
       const output = await healthcheck()
-      expect(output).toEqual('OK')
+      expect(output).toEqual('UP')
```
Comment (andrewrlee) [inline server/data/healthCheck.test.ts:32]: new health endpoint shows UP rather than OK to match spring boot

File: server/data/index.ts [hunk 11]
```diff
@@ -1,13 +1,3 @@
-/* eslint-disable import/first */
-/*
- * Do appinsights first as it does some magic instrumentation work, i.e. it affects other 'require's
- * In particular, applicationinsights automatically collects bunyan logs
- */
-import { initialiseAppInsights, buildAppInsightsClient } from '../utils/azureAppInsights'
-
-initialiseAppInsights()
```
Comment (andrewrlee) [inline server/data/index.ts:8]: This is not needed as instrumentation happens in `server.ts` before anything
(this probably wasn't required before either)

**Review by nishanthmahasamudram — APPROVED** [review]
LGTM

**Pull request merged**